build.gradle.kts

```diff
@@ -33,8 +33,10 @@ val compileSdk = 29
 
 val publishToArtifactoryTask = tasks.register<Task>("publishToArtifactory") {
     group = "publication"
-    requireNotNull(artifactoryUrl) {
-        "Property artifactoryUrl is required for publishing"
+    doLast {
+        requireNotNull(artifactoryUrl) {
+            "Property artifactoryUrl is required for publishing"
+        }
     }
```

Comment:
Again... It seems we need some custom logic to check gradle properties in configuration.

Naive implementation in a configuration block breaks IDE sync.
(even when "Do not build Gradle task list during Gradle sync" is enabled)

But we want to check it before task execution.

Reply:
Can you explain how does configuration block break sync?

Reply:
As I understand the stacktrace from IDE, it creates greedily all tasks internally by tooling API.